openedx/openedx-platform
# SE-2176 Fix elem not selected if id contains special chars

If the id of a `.formulaequationinput input` element contains a special
character, then the selector for $preview was silently failing to match
the element, because no escaping was happening.

This fixes the issue by escaping the id before passing to the jQuery
selector function. CSS.escape is the ideal method, but this isn't
present in IE or Edge, so we use a fallback borrowed from the new
jQuery.escapeSelector method. (If this is a common thing, would suggest adding a global CSS.escape polyfill instead of a local fallback)

This also fixes bugs in tests for the xsslint tool that were causing tests to fail. The `javascript-interpolate` lint rule was removed altogether since it was deemed no longer necessary.


**JIRA tickets**: [OSPR-4082](https://openedx.atlassian.net/browse/OSPR-4082)


**Dependencies**: None

**Sandbox URL**: TBD - sandbox is being provisioned.

**Merge deadline**: "None"

**Testing instructions**:

- Load a formula equation capa problem xblock from blockstore (ie. where the xblock ID doesn't go through the deprecated [html_id](https://github.com/edx/opaque-keys/blob/ac2b44c6228898ffb0d188b114465248aa72f756/opaque_keys/edx/locator.py#L948-L961) function)
- verify that everything loads correctly, the mathjax preview is rendered and updates with the text input, and the spinner shows/hides as expected
- also check that the xsslint test changes make sense


**Reviewers**

**Settings**
```yaml
```
